x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/flyout_components/add_to_lists_table/index.tsx

```diff
@@ -0,0 +1,140 @@
+import * as i18n from './translations';
+import { getSharedListsTableColumns } from '../utils';
+import { useFindExceptionListReferences } from '../../../logic/use_find_references';
+import type { ExceptionListRuleReferencesSchema } from '../../../../../../common/detection_engine/schemas/response';
+
+interface ExceptionsAddToListsComponentProps {
+  /**
+   * Normally if there's no sharedExceptionLists, this opition is disabled, however,
+   * when adding an exception item from the exception lists management page, there is no
+   * list or rule to go off of, so user can select to add the exception to any rule or to any
+   * shared list.
+   */
+  showAllSharedLists: boolean;
+  /* Shared exception lists to display as options to add item to */
+  sharedExceptionLists: ListArray;
+  onListSelectionChange?: (listsSelectedToAdd: ExceptionListSchema[]) => void;
+}
+
+const ExceptionsAddToListsComponent: React.FC<ExceptionsAddToListsComponentProps> = ({
+  showAllSharedLists,
+  sharedExceptionLists,
+  onListSelectionChange,
+}): JSX.Element => {
+  const listsToFetch = useMemo(() => {
+    return showAllSharedLists ? [] : sharedExceptionLists;
+  }, [showAllSharedLists, sharedExceptionLists]);
+  const [listsToDisplay, setListsToDisplay] = useState<ExceptionListRuleReferencesSchema[]>([]);
+  const [pagination, setPagination] = useState({ pageIndex: 0 });
+  const [message, setMessage] = useState<JSX.Element | string | undefined>(
+    <EuiLoadingContent lines={4} data-test-subj="exceptionItemListsTableLoading" />
+  );
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const [isLoadingReferences, referenceFetchError, ruleReferences, fetchReferences] =
+    useFindExceptionListReferences();
+
+  useEffect(() => {
+    if (fetchReferences != null) {
+      const listsToQuery: FindRulesReferencedByExceptionsListProp[] = !listsToFetch.length
+        ? [{ namespaceType: 'single' }, { namespaceType: 'agnostic' }]
+        : listsToFetch.map(({ id, list_id: listId, namespace_type: namespaceType }) => ({
+            id,
+            listId,
+            namespaceType,
+          }));
+      fetchReferences(listsToQuery);
+    }
+  }, [listsToFetch, fetchReferences]);
+
+  useEffect(() => {
+    if (referenceFetchError) {
+      setError(i18n.REFERENCES_FETCH_ERROR);
+    } else if (isLoadingReferences) {
+      setMessage(<EuiLoadingContent lines={4} data-test-subj="exceptionItemListsTableLoading" />);
+    } else if (ruleReferences != null) {
+      const lists: ExceptionListRuleReferencesSchema[] = [];
+      for (const [_, value] of Object.entries(ruleReferences)) {
+        if (value.type === ExceptionListTypeEnum.DETECTION) {
+          lists.push(value);
+        }
+      }
+
+      setMessage(undefined);
+      setListsToDisplay(lists);
+    }
+  }, [isLoadingReferences, referenceFetchError, ruleReferences, showAllSharedLists]);
```

Comment:
very nit: 
what about
```
if(referenceFetchError)
  return setError(i18n.REFERENCES_FETCH_ERROR);
 if (isLoadingReferences) {
    return  setMessage(<EuiLoadingContent lines={4} data-test-subj="exceptionItemListsTableLoading" />);
    } 
 if (!ruleReferences ) return;
    const lists: ExceptionListRuleReferencesSchema[] = [];
    for (const [_, value] of Object.entries(ruleReferences)) 
      if (value.type === ExceptionListTypeEnum.DETECTION) 
        lists.push(value);

    setMessage(undefined);
    setListsToDisplay(lists);
  

```

Reply:
No preference for me - happy to change! Curious, is there an advantage in what you're suggesting?

Reply:
sure thanks :), One of the best practices to make your code cleaner in JS and in many languages is to Avoid `if-else` hell.
This [post](https://www.linkedin.com/pulse/avoid-if-else-hell-javascript-afsal-m-h/) shows some replacements 

I also read before that the `if-else` affects the compiled code branching which makes it harder to test your code very well. I will try to find the resource.